src/Framework/PaymentGateways/Commands/PaymentCommand.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Give\Framework\PaymentGateways\Commands;
+
+/***
+ * @unreleased
+ */
+abstract class PaymentCommand implements GatewayCommand
+{
+    /**
+     * The Gateway Transaction / Charge Record ID
+     *
+     * @var string|null
+     */
+    public $gatewayTransactionId;
+
+    /**
+     * Notes to be added to the payment.
+     *
+     * @var array|string[]
+     */
+    public $paymentNotes = [];
+
+    /**
+     * @param  string|null  $gatewayTransactionId
+     * @return static
+     */
+    public static function make($gatewayTransactionId = null)
+    {
+        return new static($gatewayTransactionId);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param  string|null  $gatewayTransactionId
+     */
+    public function __construct($gatewayTransactionId = null)
+    {
+        $this->gatewayTransactionId = $gatewayTransactionId;
+    }
+
+    /**
+     * @param  string|string[]  ...$paymentNotes
+     * @return $this
+     */
+    public function withNote(...$paymentNotes)
+    {
+        $this->paymentNotes = $paymentNotes;
+        return $this;
+    }
```

Comment:
@kjohnson This can simply be `notes`, same as below. The `withX` convention, in my experience, is intended for immutable object changes. That is, `withNote` would clone the current object, apply the notes to the new object, and then return the new object.

Reply:
Yeah, I hadn't thought about that 👍🏻 Resolved by d4ea6da89.